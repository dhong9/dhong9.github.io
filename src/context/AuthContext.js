/**
=========================================================
* Danyo-1.2
=========================================================

* Copyright 2024 Danyo (https://www.danyo.tech)

Coded by www.danyo.tech

=========================================================

* The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
*/

import { createContext, useState, useEffect } from "react";
import { googleLogout, useGoogleLogin } from "@react-oauth/google";
import { useNavigate } from "react-router-dom";
import axios from "axios";

// Services
import {
  addAccount,
  getUser,
  loginAccount,
  updateAccount,
  deleteAccount,
} from "services/accountsService";
import { getGoogleUser } from "services/googleService";

// prop-types is a library for typechecking of props
import PropTypes from "prop-types";

const AuthContext = createContext();

export default AuthContext;

export const AuthProvider = ({ children }) => {
  // Getting auth token priority
  // 1. Session storage
  // 2. Local storage
  const sessionToken = sessionStorage.getItem("authTokens");
  const localToken = localStorage.getItem("authTokens");
  const [authTokens, setAuthTokens] = useState(() => sessionToken || localToken || null);
  const [user, setUser] = useState(null);
  const [profile, setProfile] = useState(null);

  const history = useNavigate();

  /**
   * Logs in a user
   * @param {string} username username or email address
   * @param {string} password password
   * @param {boolean} rememberMe flag to remember user or not
   * @param {Function} success success callback
   * @param {Function} error error callback
   */
  const loginUser = (username, password, rememberMe, success, error) => {
    loginAccount(
      /^[\w-.]+@([\w-]+\.)+[\w-]{2,4}$/.test(username)
        ? { email: username, password }
        : { username, password },
      (response) => {
        // Unpack token data
        const { auth_token } = response.data;
        axios.defaults.headers.common["Authorization"] = "Token " + auth_token;

        // If user wants to be remembered,
        // save their token in a local cache
        if (rememberMe) {
          localStorage.setItem("authTokens", auth_token);
        }
        // If not,
        // just maintain their token until they close their session
        else {
          sessionStorage.setItem("authTokens", auth_token);
        }

        // Navigate to home page and show success status
        history("/");
        success();

        // Get user data
        getUser(({ data }) => {
          // data {
          //     email: ...,
          //     username: ...
          // }
          if (rememberMe) {
            localStorage.setItem("user", data);
          } else {
            sessionStorage.setItem("user", data);
          }
        }, console.error);
      },
      error
    );
  };

  /**
   * Adds user to database
   * @param {string} email email address
   * @param {string} username username
   * @param {string} password password
   * @param {Function} success success callback
   * @param {Function} error error callback
   */
  const registerUser = (email, username, password, success, error) => {
    addAccount(
      {
        email,
        username,
        password,
      },
      () => {
        // If the user successfully registered,
        // take them so sign in page and display success message
        history("/pages/authentication/sign-in");
        success();
      },
      error
    );
  };

  const updateUser = (id, email, username, password, password2, success, error) => {
    updateAccount(
      id,
      {
        email,
        username,
        password,
        password2,
      },
      success,
      error
    );
  };

  const logoutUser = () => {
    googleLogout();

    // Cleared saved data
    localStorage.removeItem("authTokens");
    sessionStorage.removeItem("authTokens");
    localStorage.removeItem("user");
    sessionStorage.removeItem("user");
    setAuthTokens(null);
    setUser(null);
    setProfile(null);

    // Go back to home page
    history("/");
  };

  const deleteUser = (id, success, error) => {
    deleteAccount(id, success, error);
  };

  // Google login functions
  const googleLogin = useGoogleLogin({
    onSuccess: (codeResponse) => {
      // data has access and refresh tokens
      setAuthTokens(codeResponse);
      setUser(codeResponse);
      sessionStorage.setItem("authTokens", JSON.stringify(codeResponse));
      history("/");
    },
    onError: (error) => {
      console.error(error);
    },
  });

  const contextData = {
    user,
    setUser,
    authTokens,
    setAuthTokens,
    registerUser,
    loginUser,
    logoutUser,
    updateUser,
    deleteUser,
    profile,
    googleLogin,
  };

  useEffect(() => {
    // Get Google user
    if (authTokens && authTokens.access_token) {
      getGoogleUser(
        authTokens.access_token,
        (res) => {
          setProfile(res.data);
        },
        console.error
      );
    }
  }, [authTokens]);

  return <AuthContext.Provider value={contextData}>{children}</AuthContext.Provider>;
};

// Typechecking props of AuthContext
AuthProvider.propTypes = {
  children: PropTypes.node,
};

AuthProvider.defaultProps = {
  children: null,
};
