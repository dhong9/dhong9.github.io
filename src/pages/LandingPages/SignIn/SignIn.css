/**
=========================================================
* Danyo-1.2
=========================================================

* Copyright 2024 Danyo (https://www.danyo.tech)

Coded by www.danyo.tech

=========================================================

* The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
*/

.socialAuthButton {
    /* Button background */
    background-color: #edeff3 !important;
    width: 20vw;
    margin-bottom: 10px;
    height: 4.5vh;
    margin-bottom: 10px;
    color: #eaebed;

    /* Button text */
    font-weight: 600 !important;
    border-color: #0148a4 !important;
    border-radius: 25px !important;
    border: 0px !important;
    font-size: 0.9vw !important;
}
