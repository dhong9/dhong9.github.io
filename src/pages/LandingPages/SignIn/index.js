/**
=========================================================
* Material Kit 2 React - v2.1.0
=========================================================

* Product Page: https://www.creative-tim.com/product/material-kit-react
* Copyright 2023 Creative Tim (https://www.creative-tim.com)

Coded by www.creative-tim.com

 =========================================================

* The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
*/

import { useState, useContext } from "react";

// react-router-dom components
import { Link } from "react-router-dom";

// @mui material components
import Card from "@mui/material/Card";
import Divider from "@mui/material/Divider";
import Switch from "@mui/material/Switch";
import Grid from "@mui/material/Grid";
import MuiLink from "@mui/material/Link";

// @mui icons
import GitHubIcon from "@mui/icons-material/GitHub";
import GoogleIcon from "@mui/icons-material/Google";

// Material Kit 2 React components
import MKBox from "components/MKBox";
import MKTypography from "components/MKTypography";
import MKInput from "components/MKInput";
import MKButton from "components/MKButton";

// Material Kit 2 React example components
import DefaultNavbar from "examples/Navbars/DefaultNavbar";
import SimpleFooter from "examples/Footers/SimpleFooter";

// DH React components
import DHSnackbar from "components/DHSnackbar";

// Material Kit 2 React page layout routes
import routes from "routes";

// Authentication
import AuthContext from "context/AuthContext";
import LoginGithub from "react-login-github";

// Images
import bgImage from "assets/images/neons_medium.png";

// prop-types is a library for typechecking of props
import PropTypes from "prop-types";

// CSS
import "./SignIn.css";

function SignInBasic({ onsuccess }) {
  const { loginUser, googleLogin } = useContext(AuthContext);
  const [rememberMe, setRememberMe] = useState(false);
  const [username, setUsername] = useState("");
  const [password, setPassword] = useState("");
  const [snackbarOpen, setSnackbarOpen] = useState(false);
  const [loginSeverity, setLoginSeverity] = useState("info");
  const [loginMessage, setLoginMessage] = useState("");
  const [formErrors, setFormErrors] = useState([]);

  const handleSetRememberMe = () => setRememberMe(!rememberMe);
  const handleSubmit = async (e) => {
    e.preventDefault();

    // Check form data
    const errors = [];
    if (!username.trim()) {
      errors.push("Username is required.");
    }
    if (!password.trim()) {
      errors.push("Password is required.");
    }
    setFormErrors(errors);

    // If form input requirements are met,
    // sign the user in
    if (!errors[0]) {
      loginUser(username, password, rememberMe, onsuccess, (error) => {
        setLoginSeverity("error");
        console.log("Login error:");
        console.error(error);

        if (error.response?.data) {
          for (const field in error.response.data) {
            setLoginMessage(error.response.data[field]);
          }
        } else {
          setLoginMessage("An unexpected error has occurred.");
          console.error(error);
        }
        setSnackbarOpen(true);
      });
    }
  };

  const handleSnackbarClose = (event, reason) => {
    if (reason !== "clickaway") {
      setSnackbarOpen(false);
    }
  };

  return (
    <>
      <DefaultNavbar routes={routes} transparent light />
      <DHSnackbar
        open={snackbarOpen}
        onClose={handleSnackbarClose}
        severity={loginSeverity}
        message={loginMessage}
      />
      <MKBox
        position="absolute"
        top={0}
        left={0}
        zIndex={1}
        width="100%"
        minHeight="100vh"
        sx={{
          backgroundImage: ({ functions: { linearGradient, rgba }, palette: { gradients } }) =>
            `${linearGradient(
              rgba(gradients.dark.main, 0.6),
              rgba(gradients.dark.state, 0.6)
            )}, url(${bgImage})`,
          backgroundSize: "cover",
          backgroundPosition: "center",
          backgroundRepeat: "no-repeat",
        }}
      />
      <MKBox px={1} width="100%" height="100vh" mx="auto" position="relative" zIndex={2}>
        <Grid container spacing={1} justifyContent="center" alignItems="center" height="100%">
          <Grid item xs={11} sm={9} md={5} lg={4} xl={3}>
            <Card>
              <MKBox
                variant="gradient"
                bgColor="info"
                borderRadius="lg"
                coloredShadow="info"
                mx={2}
                mt={-3}
                p={2}
                mb={1}
                textAlign="center"
              >
                <MKTypography variant="h4" fontWeight="medium" color="white" mt={1}>
                  Sign in
                </MKTypography>
              </MKBox>
              <MKBox pt={4} pb={3} px={3}>
                <LoginGithub
                  clientId="fad22315b080996a7aaa"
                  onSuccess={console.log}
                  onFailure={console.error}
                  buttonText={
                    <MKTypography component={MuiLink} href="#" variant="body1">
                      <GitHubIcon color="inherit" />
                      <span>Sign in with GitHub</span>
                    </MKTypography>
                  }
                  className="socialAuthButton"
                />
                <MKButton className="socialAuthButton" onClick={googleLogin}>
                  <MKTypography component={MuiLink} href="#" variant="body1">
                    <GoogleIcon color="inherit" />
                    <span>Sign in with Google</span>
                  </MKTypography>
                </MKButton>
                <Divider variant="middle">
                  <MKTypography variant="h6" fontWeight="medium">
                    OR
                  </MKTypography>
                </Divider>
                <MKBox component="form" role="form">
                  <MKBox mb={2}>
                    <MKInput
                      type="text"
                      label="Username"
                      onChange={(e) => setUsername(e.target.value)}
                      fullWidth
                    />
                  </MKBox>
                  <MKBox mb={2}>
                    <MKInput
                      type="password"
                      label="Password"
                      onChange={(e) => setPassword(e.target.value)}
                      fullWidth
                    />
                  </MKBox>
                  <MKBox display="flex" alignItems="center" ml={-1}>
                    <Switch checked={rememberMe} onChange={handleSetRememberMe} />
                    <MKTypography
                      variant="button"
                      fontWeight="regular"
                      color="text"
                      onClick={handleSetRememberMe}
                      sx={{ cursor: "pointer", userSelect: "none", ml: -1 }}
                    >
                      &nbsp;&nbsp;Remember me
                    </MKTypography>
                  </MKBox>
                  {/* Show any form errors */}
                  {formErrors[0] ? (
                    <MKBox display="flex" alignItems="center" ml={-1}>
                      <ul style={{ color: "red", fontSize: "10pt" }}>
                        {formErrors.map((error, i) => (
                          <li key={i}>{error}</li>
                        ))}
                      </ul>
                    </MKBox>
                  ) : (
                    <div></div>
                  )}
                  <MKBox mt={4} mb={1}>
                    <MKButton variant="gradient" color="info" onClick={handleSubmit} fullWidth>
                      Sign In
                    </MKButton>
                  </MKBox>
                  <MKBox mt={3} mb={1} textAlign="center">
                    <MKTypography variant="button" color="text">
                      Don&apos;t have an account?{" "}
                      <MKTypography
                        component={Link}
                        to="/pages/authentication/sign-up"
                        variant="button"
                        color="info"
                        fontWeight="medium"
                        textGradient
                      >
                        Sign up
                      </MKTypography>
                    </MKTypography>
                  </MKBox>
                  <Divider variant="middle" />
                  <MKBox mt={3} mb={1} textAlign="center">
                    <MKTypography
                      component={Link}
                      to="/pages/authentication/forgot"
                      variant="button"
                      color="primary"
                      fontWeight="medium"
                      textGradient
                    >
                      Forgot password?
                    </MKTypography>
                  </MKBox>
                </MKBox>
              </MKBox>
            </Card>
          </Grid>
        </Grid>
      </MKBox>
      <MKBox width="100%" position="absolute" zIndex={2} bottom="1.625rem">
        <SimpleFooter light />
      </MKBox>
    </>
  );
}

export default SignInBasic;

// Typechecking props of SignOutPage
SignInBasic.propTypes = {
  onsuccess: PropTypes.func,
};

SignInBasic.defaultProps = {
  onsuccess: null,
};
