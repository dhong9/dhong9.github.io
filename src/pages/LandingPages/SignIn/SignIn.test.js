// React testing libraries
import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";

// Material Kit 2 React themes
import { ThemeProvider } from "@mui/material/styles";
import theme from "assets/theme";

// Authentication
import axios from "axios";
import AuthContext, { AuthProvider } from "context/AuthContext";

// Axios
import MockAdapter from "axios-mock-adapter";

// Component to test
import SignIn from "pages/LandingPages/SignIn";

jest.mock("services/baseService", () => ({
  getRequest: jest.fn(),
  postRequest: jest.fn(),
}));

// Setup axios mock
const mock = new MockAdapter(axios);

mock.onPost("accounts/token/").reply(200, {});

// Define Mocks
jest.mock("react-monaco-editor", () => {
  const { forwardRef } = jest.requireActual("react");
  return {
    __esModule: true,
    default: forwardRef(() => <div>Mock Editor</div>),
  };
});
jest.mock("draft-convert", () => {
  return {
    convertFromHTML: jest.fn(),
    convertToHTML: jest.fn(),
    convertToRaw: jest.fn(),
  };
});
jest.mock("react-router-dom", () => ({
  Link: jest.fn(({ to, children }) => <a href={to}>{children}</a>),
  useNavigate: jest.fn,
}));

describe("SignIn", () => {
  it("renders", () => {
    const contextData = {
      loginUser: jest.fn(),
    };
    const { container } = render(
      <AuthContext.Provider value={contextData}>
        <AuthProvider>
          <ThemeProvider theme={theme}>
            <SignIn />
          </ThemeProvider>
        </AuthProvider>
      </AuthContext.Provider>
    );
    
    expect(container).toMatchSnapshot();
  });
});
