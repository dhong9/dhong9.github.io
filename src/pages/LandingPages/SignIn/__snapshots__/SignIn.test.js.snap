// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn renders 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1rujksk-MuiContainer-root"
  >
    <div
      class="MuiBox-root css-t25fv4"
    >
      <div
        class="MuiBox-root css-1dyncq9"
      >
        <a
          href="/"
        >
          <span
            class="MuiTypography-root MuiTypography-button css-n07g5z-MuiTypography-root"
          >
            Daniel Hong Portfolio
          </span>
        </a>
        <div
          class="MuiBox-root css-k0f0w7"
        >
          <a
            href="/doughnutRider"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-incl6n-MuiTypography-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-eimhud-MuiSvgIcon-root"
                data-testid="DonutSmallIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11 9.16V2c-5 .5-9 4.79-9 10s4 9.5 9 10v-7.16c-1-.41-2-1.52-2-2.84s1-2.43 2-2.84zM14.86 11H22c-.48-4.75-4-8.53-9-9v7.16c1 .3 1.52.98 1.86 1.84zM13 14.84V22c5-.47 8.52-4.25 9-9h-7.14c-.34.86-.86 1.54-1.86 1.84z"
                />
              </svg>
            </p>
            <span
              class="MuiTypography-root MuiTypography-button css-1vt2rd0-MuiTypography-root"
            >
              doughnut rider
            </span>
            <p
              class="MuiTypography-root MuiTypography-body2 css-174uyvw-MuiTypography-root"
            >
              <span
                aria-hidden="true"
                class="material-icons-round notranslate MuiIcon-root MuiIcon-fontSizeInherit css-x5zaln-MuiIcon-root"
              />
            </p>
          </a>
          <div
            class="MuiBox-root css-bgea9c"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-incl6n-MuiTypography-root"
            >
              <span
                aria-hidden="true"
                class="material-icons-round notranslate MuiIcon-root MuiIcon-fontSizeInherit css-1neevca-MuiIcon-root"
              >
                view_day
              </span>
            </p>
            <span
              class="MuiTypography-root MuiTypography-button css-1vt2rd0-MuiTypography-root"
            >
              sections
            </span>
            <p
              class="MuiTypography-root MuiTypography-body2 css-174uyvw-MuiTypography-root"
            >
              <span
                aria-hidden="true"
                class="material-icons-round notranslate MuiIcon-root MuiIcon-fontSizeInherit css-x5zaln-MuiIcon-root"
              >
                keyboard_arrow_down
              </span>
            </p>
          </div>
          <a
            class="MuiBox-root css-bgea9c"
            href="https://github.com/dhong9/dhong9.github.io"
            rel="noreferrer"
            target="_blank"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-incl6n-MuiTypography-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-eimhud-MuiSvgIcon-root"
                data-testid="GitHubIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 1.27a11 11 0 00-3.48 21.46c.55.09.73-.28.73-.55v-1.84c-3.03.64-3.67-1.46-3.67-1.46-.55-1.29-1.28-1.65-1.28-1.65-.92-.65.1-.65.1-.65 1.1 0 1.73 1.1 1.73 1.1.92 1.65 2.57 1.2 3.21.92a2 2 0 01.64-1.47c-2.47-.27-5.04-1.19-5.04-5.5 0-1.1.46-2.1 1.2-2.84a3.76 3.76 0 010-2.93s.91-.28 3.11 1.1c1.8-.49 3.7-.49 5.5 0 2.1-1.38 3.02-1.1 3.02-1.1a3.76 3.76 0 010 2.93c.83.74 1.2 1.74 1.2 2.94 0 4.21-2.57 5.13-5.04 5.4.45.37.82.92.82 2.02v3.03c0 .27.1.64.73.55A11 11 0 0012 1.27"
                />
              </svg>
            </p>
            <span
              class="MuiTypography-root MuiTypography-button css-1vt2rd0-MuiTypography-root"
            >
              github
            </span>
            <p
              class="MuiTypography-root MuiTypography-body2 css-174uyvw-MuiTypography-root"
            >
              <span
                aria-hidden="true"
                class="material-icons-round notranslate MuiIcon-root MuiIcon-fontSizeInherit css-x5zaln-MuiIcon-root"
              />
            </p>
          </a>
        </div>
        <div
          class="MuiBox-root css-13l63k6"
        />
        <div
          class="MuiBox-root css-9skpnh"
        >
          <span
            aria-hidden="true"
            class="material-icons-round notranslate MuiIcon-root MuiIcon-fontSizeDefault css-75zff6-MuiIcon-root"
          >
            menu
          </span>
        </div>
      </div>
      <div
        class="MuiBox-root css-yhb49f"
      />
    </div>
  </div>
  <div
    class="MuiBox-root css-woio14"
  />
  <div
    class="MuiBox-root css-1adool0"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-9svixu-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-11 MuiGrid-grid-sm-9 MuiGrid-grid-md-5 MuiGrid-grid-lg-4 MuiGrid-grid-xl-3 css-1auz1p1-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1h15ts-MuiPaper-root-MuiCard-root"
        >
          <div
            class="MuiBox-root css-15m7mkq"
          >
            <h4
              class="MuiTypography-root MuiTypography-h4 css-o30d1n-MuiTypography-root"
            >
              Sign in
            </h4>
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-12yok62-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-veqwn4-MuiGrid-root"
              >
                <a
                  class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone MuiTypography-root MuiTypography-body1 css-t0kujj-MuiTypography-root-MuiLink-root-MuiTypography-root"
                  href="#"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-eimhud-MuiSvgIcon-root"
                    data-testid="FacebookIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M5 3h14a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2m13 2h-2.5A3.5 3.5 0 0 0 12 8.5V11h-2v3h2v7h3v-7h3v-3h-3V9a1 1 0 0 1 1-1h2V5z"
                    />
                  </svg>
                </a>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-veqwn4-MuiGrid-root"
              >
                <a
                  class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone MuiTypography-root MuiTypography-body1 css-t0kujj-MuiTypography-root-MuiLink-root-MuiTypography-root"
                  href="#"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-eimhud-MuiSvgIcon-root"
                    data-testid="GitHubIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 1.27a11 11 0 00-3.48 21.46c.55.09.73-.28.73-.55v-1.84c-3.03.64-3.67-1.46-3.67-1.46-.55-1.29-1.28-1.65-1.28-1.65-.92-.65.1-.65.1-.65 1.1 0 1.73 1.1 1.73 1.1.92 1.65 2.57 1.2 3.21.92a2 2 0 01.64-1.47c-2.47-.27-5.04-1.19-5.04-5.5 0-1.1.46-2.1 1.2-2.84a3.76 3.76 0 010-2.93s.91-.28 3.11 1.1c1.8-.49 3.7-.49 5.5 0 2.1-1.38 3.02-1.1 3.02-1.1a3.76 3.76 0 010 2.93c.83.74 1.2 1.74 1.2 2.94 0 4.21-2.57 5.13-5.04 5.4.45.37.82.92.82 2.02v3.03c0 .27.1.64.73.55A11 11 0 0012 1.27"
                    />
                  </svg>
                </a>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-veqwn4-MuiGrid-root"
              >
                <a
                  class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone MuiTypography-root MuiTypography-body1 css-t0kujj-MuiTypography-root-MuiLink-root-MuiTypography-root"
                  href="#"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-eimhud-MuiSvgIcon-root"
                    data-testid="GoogleIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12.545,10.239v3.821h5.445c-0.712,2.315-2.647,3.972-5.445,3.972c-3.332,0-6.033-2.701-6.033-6.032s2.701-6.032,6.033-6.032c1.498,0,2.866,0.549,3.921,1.453l2.814-2.814C17.503,2.988,15.139,2,12.545,2C7.021,2,2.543,6.477,2.543,12s4.478,10,10.002,10c8.396,0,10.249-7.85,9.426-11.748L12.545,10.239z"
                    />
                  </svg>
                </a>
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root css-1iu09ww"
          >
            <form
              class="MuiBox-root css-1ircn5c"
              role="form"
            >
              <div
                class="MuiBox-root css-16pyyvj"
              >
                <div
                  class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1c3j9he-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-uk4jy3-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="false"
                    for=":r0:"
                    id=":r0:-label"
                  >
                    Username
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-12n1zae-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input css-5mmmz-MuiInputBase-input-MuiOutlinedInput-input"
                      id=":r0:"
                      type="text"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-135f4hd-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-yjsfm1"
                      >
                        <span>
                          Username
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
              <div
                class="MuiBox-root css-16pyyvj"
              >
                <div
                  class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1c3j9he-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-uk4jy3-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="false"
                    for=":r1:"
                    id=":r1:-label"
                  >
                    Password
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-12n1zae-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input css-5mmmz-MuiInputBase-input-MuiOutlinedInput-input"
                      id=":r1:"
                      type="password"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-135f4hd-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-yjsfm1"
                      >
                        <span>
                          Password
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
              <div
                class="MuiBox-root css-oh6zk8"
              >
                <span
                  class="MuiSwitch-root MuiSwitch-sizeMedium css-tnsvld-MuiSwitch-root"
                >
                  <span
                    class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-19fdyy-MuiButtonBase-root-MuiSwitch-switchBase"
                  >
                    <input
                      class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
                      type="checkbox"
                    />
                    <span
                      class="MuiSwitch-thumb css-11rehke-MuiSwitch-thumb"
                    />
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </span>
                  <span
                    class="MuiSwitch-track css-1jtflaz-MuiSwitch-track"
                  />
                </span>
                <span
                  class="MuiTypography-root MuiTypography-button css-6p3i9r-MuiTypography-root"
                >
                    Remember me
                </span>
              </div>
              <div />
              <div
                class="MuiBox-root css-1id64jh"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-1vl7gcm-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="button"
                >
                  sign in
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
              <div
                class="MuiBox-root css-ne2rd"
              >
                <span
                  class="MuiTypography-root MuiTypography-button css-juj9v8-MuiTypography-root"
                >
                  Don't have an account?
                   
                  <a
                    href="/pages/authentication/sign-up"
                  >
                    Sign up
                  </a>
                </span>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiBox-root css-1fw6uxb"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1rujksk-MuiContainer-root"
    >
      <div
        class="MuiBox-root css-10nhfm9"
      >
        <div
          class="MuiBox-root css-8ulqt3"
        >
          © 
          2023
          , made with
          <div
            class="MuiBox-root css-eqjv2l"
          >
            <span
              aria-hidden="true"
              class="material-icons-round notranslate MuiIcon-root MuiIcon-fontSizeInherit css-1neevca-MuiIcon-root"
            >
              favorite
            </span>
          </div>
          by
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-9lisvk-MuiTypography-root-MuiLink-root"
            href="https://www.creative-tim.com/"
            target="_blank"
          >
            <span
              class="MuiTypography-root MuiTypography-button css-1do7hv6-MuiTypography-root"
            >
               
              Creative Tim
               
            </span>
          </a>
          for a better web.
        </div>
        <ul
          class="MuiBox-root css-9ecu6"
        >
          <li
            class="MuiBox-root css-jljawk"
          >
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-9lisvk-MuiTypography-root-MuiLink-root"
              href="https://www.creative-tim.com/"
              target="_blank"
            >
              <span
                class="MuiTypography-root MuiTypography-button css-1b76ia3-MuiTypography-root"
              >
                Creative Tim
              </span>
            </a>
          </li>
          <li
            class="MuiBox-root css-mnv3sn"
          >
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-9lisvk-MuiTypography-root-MuiLink-root"
              href="https://www.creative-tim.com/presentation"
              target="_blank"
            >
              <span
                class="MuiTypography-root MuiTypography-button css-1b76ia3-MuiTypography-root"
              >
                About Us
              </span>
            </a>
          </li>
          <li
            class="MuiBox-root css-mnv3sn"
          >
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-9lisvk-MuiTypography-root-MuiLink-root"
              href="https://www.creative-tim.com/blog"
              target="_blank"
            >
              <span
                class="MuiTypography-root MuiTypography-button css-1b76ia3-MuiTypography-root"
              >
                Blog
              </span>
            </a>
          </li>
          <li
            class="MuiBox-root css-1tpxxhu"
          >
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-9lisvk-MuiTypography-root-MuiLink-root"
              href="https://www.creative-tim.com/license"
              target="_blank"
            >
              <span
                class="MuiTypography-root MuiTypography-button css-1b76ia3-MuiTypography-root"
              >
                License
              </span>
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
